### DISCUSSION TITLE
[Enhancement] InstructP2P

### DISCUSSION THREAD

**Discussion by alexbofa** [Mikubill/sd-webui-controlnet#2066] [Ideas]
Hello, can **InstructP2P** do the same thing as **Reference only, Recolor, Revision**?
Remove the preprocessor and leave only the model so that there is no confusion?
![Screen-1866](https://github.com/Mikubill/sd-webui-controlnet/assets/58225118/3166c21d-961e-4c82-a913-ef11e760a33f)
![Screen-1867](https://github.com/Mikubill/sd-webui-controlnet/assets/58225118/866b457f-af52-46a3-97b4-caad2f16e854)

**Comment by revolverocelot1** [comment]
thats fair
